### PR TITLE
Preprocess remaster data

### PR DESCRIPTION
**Pull request by tunczyk101**

### CODE REVIEW

File: training/creating_dataset.py [hunk 1]
```diff
@@ -5,6 +5,11 @@
 import numpy as np
 import pandas as pd
 
+from training.constants import ORDERED_CHARACTERISTICS_FULL
+
+
+MAX_SPELL_LVL = 9
+"""maximum sells' level"""
```
Comment (j-adamczyk) [inline training/creating_dataset.py:12]: I don't think this comment is necessary

File: training/creating_dataset.py [hunk 2]
```diff
@@ -187,7 +192,7 @@ def get_nr_of_spells_with_lvl(items_list: list[dict], spell_level: int) -> int:
     spells = [
         i
         for i in items_list
-        if i["type"] == "spell" and i["system"]["category"]["value"] == "spell"
+        if i["type"] == "spell"  # and i["system"]["category"]["value"] == "spell"
```
Comment (j-adamczyk) [inline training/creating_dataset.py:195]: Remove commented out code if it's not necessary

File: training/save_monsters_to_books_files.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import json
+import os
+
+from training.constants import DATASET_FILES
+
+
+if __name__ == "__main__":
+    base_path = f"{os.getcwd()}/../pathfinder_2e_remaster_data"
```
Comment (j-adamczyk) [inline training/save_monsters_to_books_files.py:8]: `os.getcwd()` is not necessary here, just `../` should work

File: training/save_monsters_to_books_files.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    original_data_folder = base_path + "/packs_json"
```
Comment (j-adamczyk) [inline training/save_monsters_to_books_files.py:9]: Use f-string

File: training/save_monsters_to_books_files.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    dataset_folders_paths = os.listdir(original_data_folder)
+    dataset_folders = [folder[:-5] for folder in DATASET_FILES]
+    dataset_folders_paths = [
+        folder for folder in dataset_folders_paths if folder in dataset_folders
+    ]
+
+    for folder in dataset_folders_paths:
+        with open(f"{base_path}/{folder}.json", "a") as bestiary_file:
+            for monster in os.listdir(f"{original_data_folder}/{folder}"):
+                if "json" in monster and monster[0] != "_":
```
Comment (j-adamczyk) [inline training/save_monsters_to_books_files.py:19]: Second part can be replaced by `and not monster.startswith("_")`

File: training/save_monsters_to_books_files.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+                    with open(f"{original_data_folder}/{folder}/{monster}") as f:
```
Comment (j-adamczyk) [inline training/save_monsters_to_books_files.py:20]: Maybe `as monster_file`?

File: training/splitting_dataset.py [hunk 7]
```diff
@@ -114,6 +115,14 @@ def split_dataframe(
...
-        return get_chronological_split_results(df, test_size)
+        X_train, X_test, y_train, y_test = get_chronological_split_results(
+            df, test_size
+        )
+    else:
+        X_train, X_test, y_train, y_test = get_random_split_results(df, test_size)
+
+    if drop_book_column:
+        X_train.drop(columns=["book"], inplace=True)
+        X_test.drop(columns=["book"], inplace=True)
```
Comment (j-adamczyk) [inline training/splitting_dataset.py:126]: Do not use `inplace` if you don't have to, replace variable instead

**Review by j-adamczyk — CHANGES_REQUESTED** [review]

File: training/save_monsters_to_books_files.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    dataset_folders_paths = os.listdir(original_data_folder)
+    dataset_folders = [folder[:-5] for folder in DATASET_FILES]
+    dataset_folders_paths = [
+        folder for folder in dataset_folders_paths if folder in dataset_folders
+    ]
+
+    for folder in dataset_folders_paths:
+        with open(f"{base_path}/{folder}.json", "a") as bestiary_file:
+            for monster in os.listdir(f"{original_data_folder}/{folder}"):
+                if "json" in monster and monster.startswith("_"):
```
Comment (j-adamczyk) [inline training/save_monsters_to_books_files.py:19]: Maybe reverse this? That way, you will have one indentation less.
```
if "json" not in monster or monster.startswith("_"):
    continue

with open(...
```

**Review by j-adamczyk — APPROVED** [review]

**Comment by tunczyk101** [comment]
remaster update

**Pull request merged**